language/stdlib/modules/libra_account.mvir

```diff
@@ -278,13 +278,27 @@ module LibraAccount {
     // Creates a new account at `fresh_address` with the `initial_balance` deducted from the
     // transaction sender's account
     public create_new_account(fresh_address: address, initial_balance: u64) acquires T {
-        create_account(copy(fresh_address));
+        let account: Self.T;
+        account = T {
+            authentication_key: AddressUtil.address_to_bytes(copy(fresh_address)),
+            balance: LibraCoin.zero(),
+            delegated_key_rotation_capability: false,
+            delegated_withdrawal_capability: false,
+            received_events: Event.new_event_handle<Self.ReceivedPaymentEvent>(),
+            sent_events: Event.new_event_handle<Self.SentPaymentEvent>(),
+            sequence_number: 0,
+        };
+        Self.save_account(copy(fresh_address), move(account));
+
         if (copy(initial_balance) > 0) {
             Self.pay_from_sender(move(fresh_address), move(initial_balance));
         }
         return;
     }
 
+    // Save an account to a given address if the address does not have an account resource yet
```

Comment:
Would it make sense to check `exists<T>` before calling `save_account` instead? That would make the behavior of `save_account` simpler--we're essentially moving some Rust logic into Move.

Reply:
we can add the `exist<T>` obviously and that could make the code more obvious too.
However I guess the comment here is throwing you off a bit (and maybe I could have rephrased it somewhat) because the check for the value non existing is there by definition in the `move_to` logic, right?
In other words `move_to` always performs the check that is not overriding something, so in a sense it does not really matter.
Thoughts?

Reply:
It is true that the value not existing is there by definition in `move_to`. I guess what was not clear to me is (a) that `save_account` is doing the equivalent of a `move_to` and (b) as a user, what error (code) I can expect to see in the case that I try to create an account that already exists. If we want to have a special error for this (which may very well be a good idea!), that's a good motivation for not having the `exist` and emitting the special error inside the VM.

Reply:
I don't know those are good points to be honest and I am not sure what to say.
Today we raise `CANNOT_WRITE_EXISTING_RESOURCE` which is the same error we would raise when attempting to write over any resource in general. That is the behavior of `move_to`.
Of course that is "inevitable" as I mention and I think you got that part.
Your question is valid though and I am not sure we want a specific error here. Also I am not sure if that error should be in the form of an `assert` or a "major status" error.
Open to suggestions

Reply:
The current behavior makes sense to me given that we implement this as `move_to` under the hood.